eclair-core/src/main/scala/fr/acinq/eclair/package.scala

```diff
@@ -74,16 +74,15 @@ package object eclair {
   def feerateKw2Byte(feeratePerKw: Long): Long = feeratePerKw / 250
 
   /**
-   * why 253 and not 250 since feerate-per-kw is feerate-per-kb / 250 and the minimum relay fee rate is 1000 satoshi/Kb ?
+   * why 253 and not 250 since feerate-per-kw should be feerate-per-kvb / 4 and the minimum relay fee rate is 1000 satoshi/Kvb ?
    *
-   * because bitcoin core uses neither the actual tx size in bytes or the tx weight to check fees, but a "virtual size"
-   * which is (3 * weight) / 4 ...
+   * because bitcoin core uses neither the actual tx size in bytes nor the tx weight to check fees, but a "virtual size"
+   * which is (weight + 3) / 4 ...
    * so we want :
    * fee > 1000 * virtual size
-   * feerate-per-kw * weight > 1000 * (3 * weight / 4)
+   * feerate-per-kw * weight > 1000 * (3 + weight) / 4
    * feerate_per-kw > 250 + 3000 / (4 * weight)
    * with a conservative minimum weight of 400, we get a minimum feerate_per-kw of 253
```

Comment:
What about this to finish clarifying?

```suggestion
   * feerate-per-kw * weight > 1000 * (3 + weight) / 4
   * feerate_per-kw > 250 + 3000 / (4 * weight)
   * with a conservative minimum weight of 400, we get feerate_per-kw > 252 (because we round up),
   which leads to a minimum feerate_per-kw of 253
```

Reply:
@darosior do you think it is better this way ? If so you can apply the suggestion and we'll merge the PR. Thanks!

Reply:
@sstone I stopped to argue because it's really a nit. But here it is anyway.

I [already said](https://github.com/ACINQ/eclair/pull/1425#issuecomment-630021388) that I don't understand the reason for the round up.
This **is not what bitcoind does** (actually, if it did you would have `247` as default here !) and it seems to just fit well to justify your computation.
I think this is wrong, and I don't want to commit this under my name. That said maybe **I am** wrong but you didn't give me any rationale for this apart from "[We round up because we round up](https://github.com/ACINQ/eclair/pull/1425#issuecomment-630170178)".

You want :
```
// The * 1000 is here because the right side constant is in sat / *kilo*vbyte
// The // is a division rounded *down*
fee * 1000 // vbyte >= 1000
fee * 1000 // ((weight + 3) * 4) >= 1000
fee * 1000 // (weight + 3) >= 250
(feerate * weight) * 1000 // (weight + 3) >= 250
```
Which is only satisfied, if `weight == 400`, by `feerate >= 253` :
```python
>>> weight = 400
>>> feerate = 250
>>> (feerate * weight) // 1000 * 1000 // (weight +3)
248
>>> feerate = 252
>>> (feerate * weight) // 1000 * 1000 // (weight +3)
248
>>> feerate = 253
>>> (feerate * weight) // 1000 * 1000 // (weight +3)
250
```

Reply:
The way I see it there is just this simple inequality we want to satisfy:

```
feerate_per-kw > 250 + 3000 / (4 * weight)
```

Which, for `weight = 400`, amounts to:

```
feerate_per-kw > 251,875
```

If we want to be safe, we should assume Bitcoin Core rounds up to 252 (and if they round down, we just took extra precautions and our tx is above the treshold, so great). And if we want to be extra safe, we should assume that they check `>` instead of `>=`, hence the `253`.

Does that make sense?

Reply:
Oh are you saying that we're computing this backwards in that explanation? And what counts is going from feerate to fee in Bitcoin Core to get something that's >= 250?

Reply:
> If we want to be safe, we should assume Bitcoin Core rounds up to 252 (and if they round down, we just took extra precautions and our tx is above the treshold, so great)

Yeah, absoutely I understand for the code taking some breathin room is a good idea (hence `>=` --> `>`). Just for the explanation it's... just wrong :/

> Oh are you saying that we're computing this backwards in that explanation?

Yep. I think the original computation is based on Rusty's explanation ([the commit that introduced it reference his PR](https://github.com/ACINQ/eclair/pull/608/commits/0b7d23cf96bc88c1b7819d49d0f69ba561dadf49)).
Unfortunately he made a typo ([this line](https://github.com/ElementsProject/lightning/pull/1251/files#diff-20e1124cb17d3a9a6bba3458c5061986R283), which is [now corrected](https://github.com/ElementsProject/lightning/blob/8d98b99bde55ad0844761b584311543c9051b6f7/bitcoin/feerate.h#L19)) and that's why I think the comment here needs some more assumptions to still barely hold.

> And what counts is going from feerate to fee in Bitcoin Core to get something that's >= 250?

Yep :-)
[Here is](https://bitcoin.stackexchange.com/a/95927/101498) what `bitcoind` does for the rationale behind my calculus.

> If that's the case, even if it's a nit I think it's worth clarifying.

I think too, does my explanation above fit ?

Reply:
Great, I finally fully understand this!
I think your explanation does make sense, it's clear to me now.
Can you apply it?

Reply:
Sure, done